Update alpine base image to support che-in-che

TylerJewell opened the pull request:
This provides a base image on alpine that includes docker 1.12 which is necessary for running docker commands from within a Che container against a host.

1: Update to alpine 3.4
2: Use base docker image for 1.12
3: Configure container to use host docker daemon

TylerJewell commented:
@JamesDrummond @eivantsov @l0rd @benoitf - with this image, I was able to start / stop / info the che-launcher from the host without having to configure or install docker.  

```
docker run -it --entrypoint=/bin/bash <image-name>
bash-4.3$ docker run --rm -t -v /var/run/docker.sock:/var/run/docker.sock eclipse/che start
```

skabashnyuk commented on alpine_jdk8/Dockerfile:
```diff
@@ -49,12 +89,19 @@ RUN apk add --no-cache --virtual=build-dependencies wget ca-certificates && \
-    mv /tmp/apache-tomcat-8.0.29 /home/user/tomcat8/ && \
-    rm -rf /home/user/tomcat8/webapps/* ** \
+    wget -q "http://archive.apache.org/dist/tomcat/tomcat-8/v8.0.29/bin/apache-tomcat-8.0.29.tar.gz" && \
+    tar -xzf "apache-tomcat-8.0.29.tar.gz" && \
```
what for we need this tomcat?

TylerJewell replied:
That is a good question.  I think it's a hangover from the ubuntu_jdk stack that we use where we add in tomcat for sample projects like spring.  If we don't want this stack to be used for that, we can probably remove Tomcat and shrink it further.

skabashnyuk commented:
I suggest to add more comments inside of the code. Sometimes it's not clear why we are downloading or doing something.

skabashnyuk commented on alpine_jdk8/Dockerfile:
```diff
@@ -6,41 +6,81 @@
-    ln -s "java-${JAVA_VERSION}-oracle" "$JAVA_HOME" && \
-    ln -s "$JAVA_HOME/bin/"* "/usr/bin/" && \
-    rm -rf "$JAVA_HOME/"*src.zip && \
+    apk add --update --no-cache openssh curl mc git subversion sudo unzip bash ca-certificates openssl && \
```
>git subversion
Is this necessary?

TylerJewell replied:
Question - is the git command line no longer necessary given that we do JGit?

It's been our policy to include all of these utilities in all of our workspaces so that SSH, git, subversion and other utilities work on the command line.  We do not have to include all of these, indeed - but it really comes down to if we want to clean up other images as well.

skabashnyuk replied:
>Question - is the git command line no longer necessary given that we do JGit?
Yes. We don't need native git anymore.

TylerJewell replied:
So it comes down to a usability question.  Do we tell users to add git manually in the terminal if they want to use it in their ws terminal?  Or do we include it for them and potentially configure it so that they are ready to do a commit & push?

JamesDrummond commented:
I actually use git to push when I have not setup github variables in che.properties. However is the extra download size worth it for most users? I don't think so.
We could provide preloaded commands that can do this for the user. Would keep them away from terminal and reduce load times for most people that do not need the terminal feature.

TylerJewell commented:
@skabashnyuk - took your recommendation.  added in comments for the various sections of the Dockerfile.  I also removed subversion, git, tomcat, and some other utilities.  It shrunk the size to 286MB.  It was originally 800MB in size.
